File: SistemaAlumnos/SistemaAlumnos/RegistroNotas.cs
namespace SistemaAlumnos.Logica;

public class RegistroNotas
{
    public RegistroNotas(Estudiante estudiante, string curso)
    {
        Estudiante = estudiante;
        Curso = curso;
    }

    public Estudiante Estudiante { get; }
    public string Curso { get; }
    public double Actividad { get; private set; }
    public double Parcial { get; private set; }
    public double Proyecto { get; private set; }

    public void IngresarNotas(double actividad, double parcial, double proyecto)
    {
        ValidarNota(actividad);
        ValidarNota(parcial);
        ValidarNota(proyecto);
        Actividad = actividad;
        Parcial = parcial;
        Proyecto = proyecto;
    }

    public double CalcularPromedio()
    {
        return (Actividad + Parcial + Proyecto) / 3;
    }

    public string ObtenerEstado()
    {
        return CalcularPromedio() >= 60 ? "Aprobado" : "Reprobado";
    }

    public void MostrarResultado()
    {
        Console.WriteLine($"{Estudiante.Nombre} | {Curso} | Promedio: {CalcularPromedio():F2} | Estado: {ObtenerEstado()}");
    }

    private static void ValidarNota(double nota)
    {
        if (nota < 0 || nota > 100)
        {
            throw new ArgumentException("Las notas deben estar entre 0 y 100.");
        }
    }
}

File: SistemaAlumnos/bluej/RegistroNotas.java
public class RegistroNotas {
    private Estudiante estudiante;
    private String curso;
    private double actividad;
    private double parcial;
    private double proyecto;

    public RegistroNotas(Estudiante estudiante, String curso) {
        this.estudiante = estudiante;
        this.curso = curso;
    }

    public void ingresarNotas(double actividad, double parcial, double proyecto) {
        validarNota(actividad);
        validarNota(parcial);
        validarNota(proyecto);
        this.actividad = actividad;
        this.parcial = parcial;
        this.proyecto = proyecto;
    }

    public double calcularPromedio() {
        return (actividad + parcial + proyecto) / 3;
    }

    public String obtenerEstado() {
        return calcularPromedio() >= 60 ? "Aprobado" : "Reprobado";
    }

    public void mostrarResultado() {
        System.out.println(estudiante.nombre + " | " + curso + " | Promedio: "
                + String.format("%.2f", calcularPromedio()) + " | Estado: " + obtenerEstado());
    }

    private void validarNota(double nota) {
        if (nota < 0 || nota > 100) {
            throw new IllegalArgumentException("Las notas deben estar entre 0 y 100.");
        }
    }
}
